tests/jupyterhub_api/test_create_session.py

```diff
@@ -0,0 +1,147 @@
+
+
+    For multiple users
+
+    python3 test_create_session.py --port 443 --users user0 user1 user2 --delay 30 --json '{"LCG-rel": "LCG_95a", "platform": "x86_64-centos7-gcc7-opt", "scriptenv": "none", "ncores": 2, "memory": 8589934592, "spark-cluster": "none"}'
+
+    """
+
+    def __init__(self, port, token, users, data, delay, base_path, verify):
+        self.ref_test_name = 'Session_Creation_test'
+        JupyterhubTest.__init__(self, port, token, base_path, verify)
+        self.data = data
+        self.users = users
+        self.delay = delay
+
+    def check_create_session(self):
+
+        self.log.write("info", "creating servers..")
+        r = ""
+        for user in self.users:
+            try:
+                r = super().call_api("post", user, data=self.data, endpoint="/server")
+            except requests.exceptions.RequestException as e:
+                self.exit = 1
+                self.log.write("error", "status code " + str(r.status_code) + " " + str(e))
+
+            else:
+                if r.status_code == 202:
+                    self.log.write("info", "Successfully requested the server")
+                    """
+                     After requesting a server it needs to respond within 30 seconds otherwise there will be 
+                     Timeout Error
+                     Example:
+        
+                     TimeoutError: Server at http://172.18.0.15:8888/user/user0/ didn't respond in 30 seconds
+                     So, even after a successful request a server may not be created
+                     
+                     maximum delay should not be more than 30s
+                     
+                    """
+
+                    time.sleep(self.delay)
+
+                    """
+                    Submit a  get request to  https://localhost:443/hub/api/users/user1, this will return the information
+                    of particular user in json format, check the value of the field 'server' if it is null, server is not active
+
+                    Example:
+
+                    json of user having active server-
+                    {"kind": "user", "name": "user1", "admin": false, "groups": [], "server": "/user/user1/", "pending": null,
+                    }
+
+                    json of user have non-active server -
+                    {"kind": "user", "name": "user3", "admin": false, "groups": [], "server": null, "pending": null}
+
+                    """
+
+                    r = super().call_api("get", user, endpoint="")
+                    if r.status_code == 200:
+                        status = r.json()
+                        server_status = status['server']
+                        if server_status:
+                            self.exit = 0
+                            self.log.write("info", user + " server successfully created " + server_status)
```

Comment:
When you convert this to the logging mechanism we asked, you probably need to add error codes (so that we have them in grafana, for example). Keep this in mind.

Reply:
Sure!